drt_1d.py

```diff
@@ -88,10 +195,50 @@ def construct_transition_matrix(num_voxels:int, voxel_size:float, sigma_s:np.nda
     for i in range(num_voxels):
         for j in range(num_voxels):
```

Comment:
Transition matrix is symmetric, so only need to do lower/upper triangle and copy.

You can also handle the diagonal directly first (without a conditional) and then handle everything up to the diagonal:
```
for j in range(i):